File: El Mamoun/Lv 1/Problem 41.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

/*
Problem 41: Write a program to read a NumberOfHours and calculates the number of weeks, and days included in that 
number
*/

namespace Problem_41
{
    internal class Program
    {
        static int ReadNumber(string Message)
        {
            int Number = 0;
            Console.WriteLine(Message);
            Number = Convert.ToInt32(Console.ReadLine());

            return Number;
        }

        static double HoursToDays(int NumberOfHours)
        {
            return (double) (NumberOfHours /24);
        }

        static double HoursToWeeks(int NumberOfHours)
        {
            return (HoursToDays(NumberOfHours )/ 24);
        }

        static void Main(string[] args)
        {

            int NumberOfHours = ReadNumber("Please Enter Of Hours?: ");
            Console.WriteLine("Total Days is: " + HoursToDays(NumberOfHours));

            Console.ReadKey();
        }
    }
}

File: El Mamoun/Lv 1/Problem 41.cpp
#include<iostream>
using namespace std;

/*
Problem 41: Write a program to read a NumberOfHours and calculates the number of weeks, and days included in that
number
*/

int ReadPostiveNumber(string message)
{
	int number{ 0 };

	do
	{

		cout << message;
		cin >> number;

	} while (number < 0);

	return number;
}

float HoursToDays(float NumOfHour)
{
	return (float) NumOfHour / 24;
}

float HoursToWeeks(float NumberOfHour)
{
	return float(HoursToDays(NumberOfHour) / 7);
}

int main()
{

	float NumberOfHours = ReadPostiveNumber("Please Enter Number of Hours? ");
	float NumberOfDays = HoursToDays(NumberOfHours);
	float NumberOfWeeks = HoursToWeeks(NumberOfHours);

	cout << "\nNumber of Hours = " << NumberOfHours << endl;
	cout << "\nNumber of Days  = " << NumberOfDays  << endl;
	cout << "\nNumber of Weeks = " << NumberOfWeeks << endl;


	return 0;
}